tests/e2e/api/coin.test.ts

```diff
@@ -0,0 +1,76 @@
+import { AptosConfig, Network, Aptos, Account, Deserializer } from "../../../src";
+import { waitForTransaction } from "../../../src/internal/transaction";
+import { RawTransaction, TransactionPayloadEntryFunction } from "../../../src/transactions/instances";
+import { TypeTagStruct } from "../../../src/transactions/typeTag/typeTag";
+import { SigningScheme } from "../../../src/types";
+
+describe("coin", () => {
+  test("it generates a transfer coin transaction with AptosCoin coin type", async () => {
+    const config = new AptosConfig({ network: Network.DEVNET });
+    const aptos = new Aptos(config);
+    const sender = Account.generate({ scheme: SigningScheme.Ed25519 });
+    const recipient = Account.generate({ scheme: SigningScheme.Ed25519 });
+    await aptos.fundAccount({ accountAddress: sender.accountAddress.toString(), amount: 100000000 });
+
+    const transaction = await aptos.transaferCoinTransaction({
+      sender,
+      recipient: recipient.accountAddress.toString(),
+      amount: 10,
+    });
+
+    const txnDeserializer = new Deserializer(transaction.rawTransaction);
+    const rawTransaction = RawTransaction.deserialize(txnDeserializer);
```

Comment:
Hmm, this is an interesting way of handling it, where you need to deserialize by passing the deserializer in.  Might be interesting if we also just provide a way to deserialize the bytes directly (if you only have that).

Reply:
I believe @xbtmatt has some thoughts about it